include/cocaine/rpc/dispatch.hpp

```diff
@@ -62,8 +64,21 @@ class basic_dispatch_t {
     // an empty optional - recurrent transition (i.e. no transition at all).
 
     virtual
-    boost::optional<dispatch_ptr_t>
-    process(const decoder_t::message_type& message, const upstream_ptr_t& upstream) const = 0;
+    auto
+    process(int id, const rpc_t& rpc) const -> boost::optional<dispatch_ptr_t> = 0;
+
+    // Observers
+
+    auto
+    name() const -> std::string;
```

Comment:
Declarations like `auto fn(args_t...) -> result_type` looks ugly and here there is no need for them.